src/main/java/io/cdap/plugin/format/MultiSQLStatementInputFormat.java

```diff
@@ -62,23 +65,30 @@ public static void setInput(Configuration hConf, MultiTableConf dbConf,
   public List<InputSplit> getSplits(JobContext context) throws IOException {
     MultiTableDBConfiguration conf = new MultiTableDBConfiguration(context.getConfiguration());
 
+    return getSplits(conf);
+  }
+
+  @VisibleForTesting
+  protected List<InputSplit> getSplits(MultiTableDBConfiguration conf) {
     List<String> sqlStatements = conf.getSqlStatements();
     List<String> tableAliases = conf.getTableAliases();
     List<InputSplit> resultSplits = new ArrayList<>();
 
-    if (tableAliases.size() == 0) {
-      tableAliases = new ArrayList<>(sqlStatements.size());
+    //Handle the case where there are not enough table aliases for all SQL statements.
+    //In this case, we use as many aliases as possible and then fall back to the default logiv.
+    if (tableAliases.size() < sqlStatements.size()) {
+      tableAliases = new LinkedList<>(tableAliases);
```

Comment:
Are you sure about LinkedList? There is only very limited number of cases where it make sense.

Reply:
Considering this will grow immediately after initializing, the ArrayList will just have to resize immediately.

Alternatively, I can just initialize the ArrayList with the known size and use AddAll.

Reply:
Yes, but AFAIR it's still faster than LinkedList :). Expanding & copying an array is reported to still be faster than maintaining a double-linked-list.